From 1e893c651d5953baf7e3e173dfe459b58cdede0f Mon Sep 17 00:00:00 2001
From: Matyas Selmeci <matyas@cs.wisc.edu>
Date: Mon, 6 Jul 2015 17:34:55 -0500
Subject: [PATCH] ASN1Sequence: use getInstance instead of casts (bc 1.47)

Casting ASN1Primitive objects to ASN1Sequence objects doesn't work;
nowadays using getInstance() instead of explicit casts is recommended.
Of course, this might just move the error from compile time to runtime...
---
 .../java/org/opensciencegrid/authz/xacml/common/X509CertUtil.java | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/main/java/org/opensciencegrid/authz/xacml/common/X509CertUtil.java b/src/main/java/org/opensciencegrid/authz/xacml/common/X509CertUtil.java
index ff98115..98d04cf 100644
--- a/src/main/java/org/opensciencegrid/authz/xacml/common/X509CertUtil.java
+++ b/src/main/java/org/opensciencegrid/authz/xacml/common/X509CertUtil.java
@@ -166,15 +166,15 @@ public class X509CertUtil {
         TBSCertificateStructure tbsCert = getUserTBSCertFromX509Chain(chain);
         //subjectDN = clientcert.getSubjectX500Principal().toString();
         //subjectDN = clientcert.getSubjectDN().toString();
         //subjectDN = X509NameHelper.toString((X509Name)clientcert.getSubjectDN());
         //subjectDN = toGlobusDN(subjectDN);
-        subjectDN = X509NameHelper.toString(tbsCert.getSubject());
+        subjectDN = X509NameHelper.toString(ASN1Sequence.getInstance(tbsCert.getSubject()));
 
 
-        //ASN1Sequence seq = (ASN1Sequence)BouncyCastleUtil.duplicate(tbsCert.getSubject().toASN1Primitive());
-        subjectDN = toGlobusString((ASN1Sequence)tbsCert.getSubject().toASN1Primitive(), omitEmail);
+        //ASN1Sequence seq = ASN1Sequence.getInstance(BouncyCastleUtil.duplicate(tbsCert.getSubject().toASN1Primitive()));
+        subjectDN = toGlobusString(ASN1Sequence.getInstance(tbsCert.getSubject().toASN1Primitive()), omitEmail);
 
         // Find End-Entity Certificate, e.g. user certificate
 
         //byte[] encoded = chain[clientcertindex].getEncoded();
         //X509Cert cert = new X509Cert(encoded);
@@ -505,11 +505,11 @@ attribute : /cms/uscms/Role=cmsprod/Capability=NULL
         while (e.hasMoreElements()) {
             ASN1Set set = (ASN1Set)e.nextElement();
 	    Enumeration ee = set.getObjects();
 	    boolean didappend = false;
 	    while (ee.hasMoreElements()) {
-		ASN1Sequence s = (ASN1Sequence)ee.nextElement();
+		ASN1Sequence s = ASN1Sequence.getInstance(ee.nextElement());
 		DERObjectIdentifier oid = (DERObjectIdentifier)s.getObjectAt(0);
 		String sym = (String) X509Name.OIDLookUp.get(oid);
         if (oid.equals(X509Name.EmailAddress) && omitEmail) {
             continue;
         }
-- 
1.8.3.1

